### ISSUE TITLE
Base rule extension: [prefer-promise-reject-errors]

### ISSUE THREAD

**Issue by DetachHead** [typescript-eslint/typescript-eslint#7687]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+base+rule+extension%22) and found none that match my proposal.
- [X] I have searched the [current extension rule list](https://typescript-eslint.io/rules/#extension-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the base rule

https://eslint.org/docs/latest/rules/prefer-promise-reject-errors

### Description

the `prefer-promise-reject-errors` rule should be extended as the base rule does not catch cases where the value is not a literal

### Fail

```typescript
const foo = "asdf"
Promise.reject(foo) // no error
```

### Pass

```typescript
const foo = new Error("asdf")
Promise.reject(foo) // no error
```

### Additional Info

originally raised #7673 and https://github.com/eslint/eslint/issues/17592 for `no-throw-literal` because i wasn't aware there was already a base rule for this

**Comment by JoshuaKGoldberg** [comment]
Heh, we've really given you a convoluted path to go down. `@typescript-eslint/no-throw-literal` -> `no-throw-literal` -> `prefer-promise-reject-errors` -> 🆕 `@typescript-eslint/prefer-promise-reject-errors`. And then taking a few weeks to respond to this latest one! Sorry for the delay.

Agreed, `prefer-promise-reject-errors` looks augmentable with type information in roughly the same way that `no-throw-literal` is. Thanks for filing! Accepting PRs. 🚀
